Source/santactl/Commands/SNTCommandFileInfo.m

```diff
@@ -355,276 +406,234 @@ - (SNTAttributeBlock)signingChain {
-
-      __block NSMutableDictionary *outputHash = [[NSMutableDictionary alloc] init];
-
-      if (key && !certIndex) {
-        SNTAttributeBlock block = fi.propertyMap[key];
-        outputHash[key] = block(fi);
-      } else if (certIndex) {
-        NSArray *signingChain = fi.signingChain(fi);
-        if (key) {
-          if ([certIndex isEqual:@(-1)]) {
-            outputHash[key] = signingChain.lastObject[key];
-          } else {
-            if (certIndex.unsignedIntegerValue - 1 < signingChain.count) {
-              outputHash[key] = signingChain[certIndex.unsignedIntegerValue - 1][key];
-            }
-          }
-        } else {
-          if ([certIndex isEqual:@(-1)]) {
-            outputHash[kSigningChain] = @[ signingChain.lastObject ?: @{} ];
-          } else {
-            NSMutableArray *indexedCert = [NSMutableArray arrayWithCapacity:signingChain.count];
-            [signingChain enumerateObjectsUsingBlock:^(id obj, NSUInteger idx, BOOL *stop) {
-              if (certIndex.unsignedIntegerValue - 1 == idx) {
-                [indexedCert addObject:obj];
-              } else {
-                [indexedCert addObject:[NSNull null]];
-              }
-            }];
-            if (indexedCert.count) outputHash[kSigningChain] = indexedCert;
-          }
-        }
-      } else {
-        NSString *sha1, *sha256;
-        [fi.fileInfo hashSHA1:&sha1 SHA256:&sha256];
-        fi.propertyMap[kSHA1] = ^id (SNTCommandFileInfo *fi) { return sha1; };
-        fi.propertyMap[kSHA256] = ^id (SNTCommandFileInfo *fi) { return sha256; };
-        [fi.propertyMap enumerateKeysAndObjectsUsingBlock:^(id key, id obj, BOOL *stop) {
-          SNTAttributeBlock block = obj;
-          outputHash[key] = block(fi);
-        }];
-      }
-      if (outputHash.count) {
-        dispatch_group_async(outputHashesGroup, outputHashesQueue, ^{
-          [outputHashes addObject:outputHash];
-        });
-      }
-    }];
-    
-    hashOperation.qualityOfService = NSQualityOfServiceUserInitiated;
-    [hashQueue addOperation:hashOperation];
-  }];
+  NSArray *filePaths = [self parseArguments:arguments];
 
-  // Wait for all the calculating threads to finish
-  [hashQueue waitUntilAllOperationsAreFinished];
+  if (!self.outputKeyList || !self.outputKeyList.count) {
+    self.outputKeyList = [[self class] fileInfoKeys];
+  }
 
-  // Clear the "Calculating ..." indicator if present
-  if (PrettyOutput()) printf("\33[2K\r");
+  // For consistency, JSON output is always returned as an array of file info objects, regardless of
+  // how many file info objects are being outputted.  So both empty and singleton result sets are
+  // still enclosed in brackets.
+  if (self.jsonOutput) printf("[\n");
+
+  NSFileManager *fm = [NSFileManager defaultManager];
+  NSString *cwd = [fm currentDirectoryPath];
+  for (NSString *path in filePaths) {
+    NSString *fullPath = [path stringByStandardizingPath];
+    if ([path characterAtIndex:0] != '/') {
```

Comment:
To catch empty strings.
if (fullPath.length && [fullPath characterAtIndex:0] != '/') {
...